src/main.rs

```diff
@@ -1,23 +1,35 @@
 use std::env;
-use std::io::{self, Write, BufWriter};
+use std::io::{self, Write};
 use std::process;
 use std::borrow::Cow;
 
-fn main() {
-    {
-        // Move all locals into a scope, so that `process::exit()` is called after all destructors.
-        let output = env::args()
-            .skip(1)
-            .next()
-            .map(|x| Cow::Owned(x + "\n"))
-            .unwrap_or(Cow::Borrowed("y\n"));
-        let stdout = io::stdout();
-        let mut buffered = BufWriter::new(stdout.lock());
-        loop {
-            if let Err(_) = buffered.write(output.as_bytes()) {
-                break;
-            }
+const BUFFER_CAPACITY: usize = 64 * 1024;
+
+fn write(output_bytes: &[u8]) {
+    let stdout = io::stdout();
+    let mut locked = stdout.lock();
+    let mut buffer = [0u8; BUFFER_CAPACITY];
+    let mut buffer_size = output_bytes.len();
+
+    let _ = (&mut buffer[..]).write(output_bytes);
+    while buffer_size < BUFFER_CAPACITY {
+        let (left, mut right) = buffer.split_at_mut(buffer_size);
+        if !locked.write_all(left).is_ok() {
```

Comment:
Did you benchmark this? How much does it affect startup time? Or is it a premature optimization?